networking/external_dns_operator/nw-control-dns-records-public-hosted-zone-azure.adoc

```diff
@@ -0,0 +1,108 @@
+$ CLIENT_SECRET=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_client_secret}} | base64 -d)
+$ RESOURCE_GROUP=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_resourcegroup}} | base64 -d)
+$ SUBSCRIPTION_ID=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_subscription_id}} | base64 -d)
+$ TENANT_ID=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_tenant_id}} | base64 -d)
+----
+
+. Login to azure with base64 decoded values:
++
+[source,terminal]
+----
+$ az login --service-principal -u "${CLIENT_ID}" -p "${CLIENT_SECRET}" --tenant "${TENANT_ID}"
+----
+
+. Get the routes to check the domain:
++
+[source,terminal]
+----
+$ oc get routes --all-namespaces | grep console
+----
++
+.Example output
+[source,terminal]
+----
+openshift-console          console             console-openshift-console.apps.test-azure.qe.azure.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.test-azure.qe.azure.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+----
+
+. Get the list of dns zones to find the one which corresponds to the previously found route's domain:
++
+[source,terminal]
+----
+$ az network dns zone list --resource-group "${RESOURCE_GROUP}"
+----
+
+. Create `ExternalDNS` CR for `route` source:
++
+[source,yaml]
+----
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-azure <1>
+spec:
+  zones:
+  - "/subscriptions/53b4f551-f0fc-4bea-8cba-11111111111/resourceGroups/test-azure1-nxkxm-rg/providers/Microsoft.Network/dnszones/test-azure.qe.azure.devcluster.openshift.com" <2>
+  provider:
+    type: Azure <3>
+  source:
+    openshiftRouteOptions: <4>
+      routerName: default <5>
+    type: OpenShiftRoute <6>
+----
+<1> defines the name of External DNS CR.
+<2> Define the zone ID.
+<3> defines the Azure DNS provider.
+<4> You can define options for the source of DNS records.
+<5> If the source is `OpenShiftRoute` then you can pass the OpenShift ingress controller name. External DNS selects the canonical hostname of that router as the target while creating CNAME record.
+<6> Defines OpenShift `route` resource as the source for the DNS records which gets created in the previously specified DNS provider.
+
+. Check the records created for OCP routes using the following command:
++
+[source,terminal]
+----
+$ az network dns record-set list -g "${RESOURCE_GROUP}"  -z test-azure.qe.azure.devcluster.openshift.com | grep console
+----
+
+.  Create a route.
++
+[source,terminal]
+----
+$ oc new-app --docker-image=openshift/hello-openshift -l app=hello-openshift
+$ oc expose service/hello-openshift -l app=hello-openshift
+----
```

Comment:
No need for this one as we already checked the records for `console` routes in the previous command.

Reply:
Addressed